backend/infrahub/events/models.py

```diff
@@ -100,3 +101,233 @@ def get_related(self) -> list[dict[str, str]]:
+            return UpdateStatus.UNMODIFIED
+        if self.value_previous is not None and self.value_current is None:
+            return UpdateStatus.REMOVED
+        if self.value_previous is None and self.value_current is not None:
+            return UpdateStatus.ADDED
+
+        return UpdateStatus.UPDATED
+
+
+###############################################################################
+# Attributes
+###############################################################################
+
+
+class CreatedAttribute(BaseModel):
+    name: str = Field(..., description="The name of the attribute")
+    value: Any = Field(..., description="The value used during creation of this attribute")
+    properties: list[Property] = Field(
+        default_factory=list, description="Properties defined for the attribute during creation"
+    )
+
+    @computed_field
+    def value_type(self) -> str:
+        """The value_type of the attribute, used to help external systems"""
+
+        # Add more types
+        if isinstance(self.value, str):
+            return "Text"
+
+        return "Boolean"
+
+
+class UpdatedAttribute(BaseModel):
+    name: str = Field(..., description="The name of the attribute")
+    value_current: Any = Field(..., description="The current value of the attribute")
+    value_previous: Any = Field(..., description="The previous value of the attribute")
+    updated_properties: list[UpdatedProperty] = Field(
+        default_factory=list, description="The properties that were updated during this update"
+    )
+
+    @computed_field
+    def value_update_status(self) -> UpdateStatus:
+        """Indicate how the peer was changed during this update"""
+        if self.value_current == self.value_previous:
+            return UpdateStatus.UNMODIFIED
+        if self.value_previous is not None and self.value_current is None:
+            return UpdateStatus.REMOVED
+        if self.value_previous is None and not self.value_current is not None:
+            return UpdateStatus.ADDED
+
+        return UpdateStatus.UPDATED
+
+    @computed_field
+    def value_type(self) -> str:
+        """The value_type of the attribute, used to help external systems
+
+        The value_type will be based on the current or previous value of this attribute
+        """
+
+        # Add more types
+        if isinstance(self.value_current, str):
+            return "Text"
+
+        return "Boolean"
+
+
+###############################################################################
+# Relationships
+###############################################################################
+
+
+class Relationship(BaseModel):
+    peer_id: str = Field(..., description="The ID of the peer on the remote end of this relationship")
+    peer_kind: str = Field(..., description="The node kind of the remote peer")
+    properties: list[Property] = Field(default_factory=list, description="The properties of this relationship")
```

Comment:
`peer` information is there, what about the other node? Is it contained within properties? Otherwise, should this class contain 2 `MutatedNode` (renamed to `Node`) instead of `peer_id` / `peer_kind`, so we also send nodes' `display_label`?

Reply:
I'd say that it depends on how much lookup work we want to do during each mutation. I.e. if we want to lookup the display label of the peer it will be additional queries to get that information. With the id and kind an external system would at least be able to find the nodes in an easy way. Since this is something that we'd always be sending it would be nice not to add too much of a performance overhead, but we also want to have information that is actually useful.